azext_iot/operations/hub.py

```diff
@@ -1377,6 +1377,7 @@ def _process_config_content(content, config_type):
     if config_type == ConfigType.edge or config_type == ConfigType.layered:
         valid_edge_key = "modulesContent"
         legacy_edge_key = "moduleContent"
+        purchase_edge_key = "modulesPurchase"
```

Comment:
We may want to consider doing client side validation of the auth type when executing, such that if the auth type is 'key' based we throw an error with auth requirements (like as shown in the help desc) to save the user a request.